### PR TITLE
B 21011 ssw page

### PR DESCRIPTION
**Pull request by antgmann**
## [B-21011](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21011) and [B-21012](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21012)

[INT PR](https://github.com/transcom/mymove/pull/13651)

## Summary
These two agility backlog items add an additional page to the SSW for AOA and Payment Packets. It's an additional shipments page to keep track of all other shipments attached to the move. It has the same look as the other pages with rows for shipment information for up to 16 additional shipments (if more are supplied, the packet will not be created).

The values that display for the shipments can depend on shipment type, so please be sure to check them thoroughly when testing.

Additionally, please feel free to try to break the PDF generation process in other ways. This task added many potential redundant values that will most often be empty, so I created a new map and added on to the existing reflection system to make things look nicer and run smoother. I'm confident in it's stability, but just in case.

> [!WARNING]  
> Commits 2110b0f507fc61abab9070342a61b71945b62036 and 0c1eb5e2f02582d40461842b4dbf3abcd7b584be on INT were merge commits required to resolve merge conflicts. If additional conflicts arise with a main PR for 21011, 21012, or [20947](https://github.com/transcom/mymove/pull/13594) , please reference these commits for resolutions.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Access the application as a service member. Create a move with many shipments including at least 1 ppm with an advance. We need to make a PPM with advance to generate the AOA, and a completed PPM for the Payment Packet.
2. Submit the move with many shipments.
3. Log in as a service counselor, approve the advance request in order to unlock the AOA packet. Add more information to other shipments as you see fit for testing.
4. Download the AOA. Ensure the packet downloads, the third page is present and looks nice with preparation date and shipment information NOT including the PPM you generated the packet for (that information is shown on the first page)
5. Ensure the PPM move details have been submitted.
6. Logged back in as a service member, upload your documents for your PPM.
7. As a service counselor, approve the documents, and generate the payment packet.
8. Ensure all data is present in it's most updated form. 
9. Ensure the watermark on orders and other documents isn't present on SSW pages.
10. Ensure the tab name in your browser is "Shipment Summary Worksheet" (used to be SSWPDFTemplate)
11. Ensure all tests passing.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
image

### CODE REVIEW

**Review by r-mettler — APPROVED** [review]

File: pkg/services/shipment_summary_worksheet.go [hunk 1]
```diff
@@ -181,10 +189,10 @@ type SSWMaxWeightEntitlement struct {
 type SSWPPMComputer interface {
```
Comment (MInthavongsay) [inline pkg/services/shipment_summary_worksheet.go:189]: The deltas of the SSWPPMComputer interface looks different than in INT PR. `FormatValuesShipmentSummaryWorksheetFormPage3` in INT PR is not in this PR. And I reading this correct?

File: pkg/services/mocks/SSWPPMComputer.go
Comment (traskowskycaci) [inline pkg/services/mocks/SSWPPMComputer.go:1]: Not seeing a // FormatValuesShipmentSummaryWorksheetFormPage3 in this file to match the main PR

File: pkg/services/shipment_summary_worksheet.go
Comment (traskowskycaci) [inline pkg/services/shipment_summary_worksheet.go:1]: Agree with Michael, seeing some diffs in this file as well

File: pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go [hunk 2]
```diff
@@ -56,17 +56,21 @@ func NewSSWPPMGenerator(pdfGenerator *paperwork.Generator) (services.SSWPPMGener
 }
 
 // FormatValuesShipmentSummaryWorksheet returns the formatted pages for the Shipment Summary Worksheet
-func (SSWPPMComputer *SSWPPMComputer) FormatValuesShipmentSummaryWorksheet(shipmentSummaryFormData services.ShipmentSummaryFormData, isPaymentPacket bool) (services.Page1Values, services.Page2Values, error) {
+func (SSWPPMComputer *SSWPPMComputer) FormatValuesShipmentSummaryWorksheet(shipmentSummaryFormData services.ShipmentSummaryFormData, isPaymentPacket bool) (services.Page1Values, services.Page2Values, services.Page3Values, error) {
```
Comment (traskowskycaci) [inline pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go:59]: This function takes in models.ShipmentSummaryFormData as the second parameter in the INT PR. The main PR here take in services.ShipmentSummaryFormData as the second param

**Review by traskowskycaci — COMMENTED** [review]
Need to add AllShipments             []MTOShipment to pkg/models/worksheet_shipment.go as part of this main PR after we merge in B-20947 to main

File: pkg/services/shipment_summary_worksheet.go [hunk 3]
```diff
@@ -136,13 +143,13 @@ type FormattedMovingExpenses struct {
...
 	FetchDataShipmentSummaryWorksheetFormData(appCtx appcontext.AppContext, _ *auth.Session, ppmShipmentID uuid.UUID) (*models.ShipmentSummaryFormData, error)
 	ComputeObligations(_ appcontext.AppContext, _ models.ShipmentSummaryFormData, _ route.Planner) (models.Obligations, error)
-	FormatValuesShipmentSummaryWorksheet(shipmentSummaryFormData models.ShipmentSummaryFormData, isPaymentPacket bool) (Page1Values, Page2Values, error)
+	FormatValuesShipmentSummaryWorksheet(shipmentSummaryFormData models.ShipmentSummaryFormData, isPaymentPacket bool) (Page1Values, Page2Values, Page3Values, error)
 	FormatShipment(ppm models.PPMShipment, weightAllotment models.SSWMaxWeightEntitlement, isPaymentPacket bool) models.WorkSheetShipment
 	FormatValuesShipmentSummaryWorksheetFormPage1(data models.ShipmentSummaryFormData, isPaymentPacket bool) (Page1Values, error)
 	FormatValuesShipmentSummaryWorksheetFormPage2(data models.ShipmentSummaryFormData, isPaymentPacket bool) (Page2Values, error)
 }
```
Comment (traskowskycaci) [inline pkg/services/shipment_summary_worksheet.go:150]: Need to add FormatValuesShipmentSummaryWorksheetFormPage3 here to match integrationTesting

File: pkg/services/shipment_summary_worksheet/shipment_summary_worksheet_test.go [hunk 4]
```diff
@@ -483,6 +484,61 @@ func (suite *ShipmentSummaryWorksheetServiceSuite) TestFormatValuesShipmentSumma
...
+		AllShipments:            move.MTOShipments,
+		ServiceMember:           serviceMember,
+		Order:                   order,
+		CurrentDutyLocation:     yuma,
+		NewDutyLocation:         fortGordon,
+		PPMRemainingEntitlement: 3000,
+		WeightAllotment:         wtgEntitlements,
+		PreparationDate:         time.Date(2019, 1, 1, 1, 1, 1, 1, time.UTC),
+		PPMShipment:             PPMShipment,
+	}
```
Comment (traskowskycaci) [inline pkg/services/shipment_summary_worksheet/shipment_summary_worksheet_test.go:520]: Missing

	mockPPMCloseoutFetcher := &mocks.PPMCloseoutFetcher{}
	sswPPMComputer := NewSSWPPMComputer(mockPPMCloseoutFetcher)


on line 796 and 797

**Review by traskowskycaci — APPROVED** [review]

File: pkg/services/shipment_summary_worksheet.go (hunk 1 above)
Comment (r-mettler) [reply pkg/services/shipment_summary_worksheet.go:143]: SSWPPMComputer interface was moved as part of B-20947 to another file which is why there were differences. Resolved them after we merged in B-20947 to main.

File: pkg/services/shipment_summary_worksheet.go (hunk 3 above)
Comment (r-mettler) [reply pkg/services/shipment_summary_worksheet.go:151]: Added

File: pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go (hunk 2 above)
Comment (r-mettler) [reply pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go]: as part of B-20947 this was moved to models/worksheet_shipment.go once B-20947 was merged into main this was resolved.

**Comment by traskowskycaci** [comment]
> Need to add AllShipments []MTOShipment to pkg/models/worksheet_shipment.go as part of this main PR after we merge in B-20947 to main

Was added, resolved

**Pull request merged**